core/network/impl/peer_view.cpp

```diff
@@ -10,22 +10,44 @@
 #include "utils/weak_macro.hpp"
 
 namespace kagome::network {
-  inline View makeView(const LazySPtr<blockchain::BlockTree> &block_tree) {
+  static constexpr size_t MAX_VIEW_HEADS = 5;
+
+  inline std::pair<View, std::vector<primitives::BlockHash>> makeView(
+      const LazySPtr<blockchain::BlockTree> &block_tree) {
     auto last_finalized = block_tree.get()->getLastFinalized().number;
+    const auto leaves = block_tree.get()->getLeavesInfo();
```

Comment:
`getLeavesInfo` returns `std::vector<BlockInfo>`, which sorts by height.
Also it never returns `info.number < last_finalized`, so loop below can be removed.
May sort and truncate `getLeavesInfo` result to simplify code.